Sources/SignalProducer.swift

```diff
@@ -1385,6 +1410,35 @@ extension SignalProducer where Error == NoError {
 		return lift { $0.timeout(after: interval, raising: error, on: scheduler) }
 	}
 
+	/// Forward events from `self` until `interval`. Then if producer isn't
+	/// completed yet, sends `event` and terminates.
+	///
+	/// - note: If `event` is a value, the returned producer will automatically
+	///         complete after sending it.
+	///
+	/// - note: If the interval is 0, the timeout will be scheduled immediately.
+	///         The signal must complete synchronously (or on a faster
+	///         scheduler) to avoid the timeout.
+	///
+	/// - precondition: `interval` must be non-negative number.
+	///
+	/// - parameters:
+	///   - event: An event to terminate with if `self` is not completed
+	///            when `interval` passes.
+	///   - interval: Number of seconds to wait for `self` to complete.
+	///   - scheduler: A scheduler to deliver error on.
+	///
+	/// - returns: A signal that sends events for at most `interval` seconds,
+	///            then, if not `completed` - sends `error` with failed event
+	///            on `scheduler`.
+	public func timeout<NewError: Swift.Error>(
+		after interval: TimeInterval,
+		sending event: SignalProducer<Value, NewError>.ProducedSignal.Event,
```

Comment:
I was thinking more for consistency: in `Signal` we can just reference `Event`, but in `SignalProducer` we need to use `ProducedSignal.Event`. I was thinking that

```
typealias Event = ProducedSignal.Event
```

would allow for consistency across the two types.